R/scale-.r

```diff
@@ -563,7 +563,7 @@ continuous_scale <- function(aesthetics, scale_name, palette, name = waiver(),
 
   position <- match.arg(position, c("left", "right", "top", "bottom"))
 
-  if (is.null(breaks) && !is_position_aes(aesthetics) && guide != "none") {
+  if (is.null(breaks) && all(is_position_aes(aesthetics)) && identical(guide, "none")) {
```

Comment:
I'm confused here. Doesn't this invert the logic for the last two checks? Wouldn't they need `!` in front?

Reply:
Opps..., sorry, I'm also confused because the previous logic seems unclear to me about its intension. Let me think. (I believe `all(is_position_aes(aesthetics))` doesn't need `!`; if there are only positional aesthetics, we don't need guides, right?)

Reply:
OK, I got it. #581 and #579 are the context; `breaks = NULL` can be used for removing the guide of a non-positional scales.